crates/tx-graph2/src/connectors/claim_contest_connector.rs

```diff
@@ -0,0 +1,439 @@
+                    n_of_n_signature,
+                    spend_path: ClaimContestSpendPath::Uncontested,
+                },
+                Ordering::Greater => panic!("Leaf index is out of bounds"),
+            }
+        }
+    }
+
+    fn spend_connector(connector: ClaimContestConnector, signer: Signer, leaf_index: usize) {
+        logging::init(LoggerConfig::new("connector-p-script-path".to_string()));
+
+        // Setup Bitcoin node
+        let mut conf = Conf::default();
+        conf.args.push("-txindex=1");
+        let bitcoind = Node::with_conf("bitcoind", &conf).unwrap();
+        let btc_client = &bitcoind.client;
+
+        // Mine until maturity
+        let funded_address = btc_client.new_address().unwrap();
+        let change_address = btc_client.new_address().unwrap();
+        let coinbase_block = btc_client
+            .generate_to_address(101, &funded_address)
+            .expect("must be able to generate blocks")
+            .0
+            .first()
+            .expect("must be able to get the blocks")
+            .parse::<BlockHash>()
+            .expect("must parse");
+        let coinbase_txid = btc_client
+            .get_block(coinbase_block)
+            .expect("must be able to get coinbase block")
+            .coinbase()
+            .expect("must be able to get the coinbase transaction")
+            .compute_txid();
+
+        // Create funding transaction
+        let funding_input = OutPoint {
+            txid: coinbase_txid,
+            vout: 0,
+        };
+
+        let coinbase_amount = Amount::from_btc(50.0).expect("must be valid amount");
+        let funding_amount = Amount::from_sat(50_000);
+        let fees = Amount::from_sat(1_000);
+
+        let input = vec![TxIn {
+            previous_output: funding_input,
+            script_sig: funded_address.script_pubkey(),
+            ..Default::default()
+        }];
+
+        let output = vec![
+            TxOut {
+                value: funding_amount,
+                script_pubkey: connector.tx_out().script_pubkey,
+            },
+            TxOut {
+                value: coinbase_amount
+                    .checked_sub(funding_amount)
+                    .unwrap()
+                    .checked_sub(fees)
+                    .unwrap(),
+                script_pubkey: change_address.script_pubkey(),
+            },
+        ];
+
+        let funding_tx = Transaction {
+            version: transaction::Version(2),
+            lock_time: absolute::LockTime::ZERO,
+            input,
+            output,
+        };
+
+        // Sign and broadcast funding transaction
+        let signed_funding_tx = btc_client
+            .call::<SignRawTransactionWithWallet>(
+                "signrawtransactionwithwallet",
+                &[json!(consensus::encode::serialize_hex(&&funding_tx))],
+            )
+            .expect("must be able to sign transaction");
+
+        assert!(signed_funding_tx.complete);
```

Comment:
It's good practice to add a message here:
```suggestion
        assert!(signed_funding_tx.complete, "funding transaction is not fully signed")
```